helm/cluster/templates/_helpers.tpl

```diff
@@ -151,4 +151,115 @@ Where `data` is the data to hash and `global` is the top level scope.
 {{- end -}}
 {{- end -}}
 {{- $registry -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{- define "cluster.internal.get-provider-integration-values" }}
+{{- $providerIntegration := dict }}
+{{- /* Case 1: template is called from the cluster chart itself */}}
+{{- if and (eq $.Chart.Name "cluster") $.Values.providerIntegration }}
+  {{- $providerIntegration = $.Values.providerIntegration }}
+{{- /* Case 2: template is called from a parent chart that passes Helm values to cluster chart in .Values.cluster */}}
+{{- else if ($.Values.cluster).providerIntegration }}
+  {{- $providerIntegration = $.Values.cluster.providerIntegration }}
+{{- end }}
+{{- /* Create $.GiantSwarm object where we put custom Giant Swarm vars */}}
+{{- if not $.GiantSwarm }}
+  {{- $_ := set $ "GiantSwarm" dict }}
+{{- end }}
+{{- /* Finally set "UseReleases" property that can be used in other templates */}}
+{{- $_ := set $.GiantSwarm "providerIntegration" $providerIntegration }}
+{{- end }}
+
+{{- define "cluster.internal.get-internal-values" }}
```

Comment:
Same like above for `cluster.internal.get-provider-integration-values` template.